### ISSUE TITLE
Rails cop idea: use bang methods when not checking return result

### ISSUE THREAD

**Issue by mockdeep** [bbatsov/rubocop#3128]
We generally prefer to have things fail loud when we're not checking the return result, so it would be great to have a rubocop rule to enforce this strategy:

```ruby
# bad
user.update(email: 'not an email') # silently fails

# good
user.update!(email: 'not an email') # => validation error

# good
if user.update(email: 'not an email')
  ...
else
  ...
end
```

Same with `save!`, `create!`, and other AR methods.
